com.microsoft.java.lsif.core/src/com/microsoft/java/lsif/core/internal/visitors/LsifVisitor.java

```diff
@@ -24,36 +48,47 @@
 import com.microsoft.java.lsif.core.internal.indexer.LsifService;
 import com.microsoft.java.lsif.core.internal.indexer.Repository;
 import com.microsoft.java.lsif.core.internal.protocol.Document;
+import com.microsoft.java.lsif.core.internal.protocol.Moniker.MonikerKind;
+import com.microsoft.java.lsif.core.internal.protocol.PackageInformation.PackageManager;
 import com.microsoft.java.lsif.core.internal.protocol.Project;
 import com.microsoft.java.lsif.core.internal.protocol.Range;
 import com.microsoft.java.lsif.core.internal.protocol.ResultSet;
 
 public class LsifVisitor extends ProtocolVisitor {
 
-	public LsifVisitor(LsifService lsif, IndexerContext context) {
+	private boolean isPublish;
+
+	public LsifVisitor(LsifService lsif, IndexerContext context, boolean isPublish) {
 		super(lsif, context);
+		this.isPublish = isPublish;
 	}
 
 	@Override
 	public boolean visit(SimpleName node) {
-		resolve(node.getStartPosition(), node.getLength(), isTypeOrMethodDeclaration(node));
+		IBinding binding = node.resolveBinding();
+		if (binding != null) {
```

Comment:
Do we know if an element does not have a binding, can we resolve other information about this element?